flair/models/biomedical_entity_linking.py

```diff
@@ -0,0 +1,988 @@
+
+        self.embed_dictionary(
+            model_name_or_path=model_name_or_path,
+            dictionary_name_or_path=dictionary_name_or_path,
+            mean_centering=mean_centering,
+            batch_size=batch_size,
+        )
+
+        return self
+
+    def load_dense_encoder(
+        self, model_name_or_path: Union[str, Path]
+    ) -> Tuple[PreTrainedModel, PreTrainedTokenizer]:
+
+        self.encoder = TransformerDocumentEmbeddings(
+            model_name_or_path, is_token_embedding=False
+        )
+
+        return self.encoder
+
+    def load_sparse_encoder(
+        self, model_name_or_path: Union[str, Path]
+    ) -> SparseWordVectorizer:
+        sparse_encoder_path = os.path.join(model_name_or_path, "sparse_encoder.pk")
+        # check file exists
+        if not os.path.isfile(sparse_encoder_path):
+            # download from huggingface hub and cache it
+            sparse_encoder_url = hf_hub_url(
+                model_name_or_path, filename="sparse_encoder.pk"
+            )
+            sparse_encoder_path = cached_download(
+                url=sparse_encoder_url,
+                cache_dir=flair.cache_root / "models" / model_name_or_path,
+            )
+
+        self.sparse_encoder = SparseWordVectorizer.load(path=sparse_encoder_path)
+
+        return self.sparse_encoder
+
+    def load_sparse_weight(self, model_name_or_path: Union[str, Path]) -> torch.Tensor:
+        sparse_weight_path = os.path.join(model_name_or_path, "sparse_weight.pt")
+        # check file exists
+        if not os.path.isfile(sparse_weight_path):
+            # download from huggingface hub and cache it
+            sparse_weight_url = hf_hub_url(
+                model_name_or_path, filename="sparse_weight.pt"
+            )
+            sparse_weight_path = cached_download(
+                url=sparse_weight_url,
+                cache_dir=flair.cache_root / "models" / model_name_or_path,
+            )
+
+        self.sparse_weight = torch.load(sparse_weight_path)
+
+        return self.sparse_weight
+
+    def get_sparse_weight(self) -> torch.Tensor:
+        assert self.sparse_weight is not None
+
+        return self.sparse_weight
+
+    def embed_sparse(
+        self, names: list, show_progress: bool = False, batch_size: int = 1024
+    ) -> np.ndarray:
+        """
+        Embedding data into sparse representations
+        :param names np.array: An array of names
+        :param show_progress bool: Whether to show a progress bar
+        :param batch_size int: The batch size for embedding
+        :returns sparse_embeds np.array: A list of sparse embeddings
+        """
+        sparse_embeds = []
+
+        if show_progress:
+            iterations = tqdm(
+                range(0, len(names), batch_size),
+                desc="Calculating sparse embeddings for dictionary",
+            )
+        else:
+            iterations = range(0, len(names), batch_size)
+
+        for start in iterations:
```

Comment:
I think that in this case we need to pass all `names` at once otherwise we get different `_vocabulary` in the vectorizer.